pkg/httpmock/stub.go

```diff
@@ -238,6 +238,20 @@ func ScopesResponder(scopes string) func(*http.Request) (*http.Response, error)
 	}
 }
 
+// StatusScopesResponder returns a response with the given status code and OAuth scopes.
+func StatusScopesResponder(status int, scopes string) func(*http.Request) (*http.Response, error) {
+	return func(req *http.Request) (*http.Response, error) {
+		return &http.Response{
+			StatusCode: status,
+			Request:    req,
+			Header: map[string][]string{
+				"X-Oauth-Scopes": {scopes},
+			},
+			Body: io.NopCloser(bytes.NewBufferString("")),
+		}, nil
+	}
+}
```

Comment:
Given how similar this is to `ScopesResponder`, should this go away and that function be updated to require providing the status code?  Or have it be a wrapper to this and passing the default status code?

My knee jerk is avoiding the wrapper and have callers' specify the exepected status code. 🤷

Reply:
Updated `ScopesResponder` to wrap `StatusScopesResponder` in 11dc6df. I think if a refactor is required to the 35 existing tests through changing `ScopesResponder`'s signature, it should be done in another PR and shouldn't block the value in this PR from being released.